### ISSUE TITLE
[BUG] Unable to connect from windows to linux using remote-schema flag

### ISSUE THREAD

**Issue by xastor** [rdiff-backup/rdiff-backup#494]
## Bug summary

Unable to connect from windows to linux using remote-schema flag.

## Version, Python, Operating System

Call `rdiff-backup --version --api-version 201` or `rdiff-backup -v9` and replace the following line with the output, repeat for each environment impacted:

Windows : 

```yaml
PS D:\repos\backup_manager> D:\repos\backup_manager\rdiff-backup-2.0.5\rdiff-backup.exe -v9
2020-12-18 12:11:10.822503 +0100  <CLIENT-10464>  Using rdiff-backup version 2.0.5
2020-12-18 12:11:10.822503 +0100  <CLIENT-10464>        with cpython D:\repos\backup_manager\rdiff-backup-2.0.5\rdiff-backup.exe version 3.7.7
2020-12-18 12:11:10.823504 +0100  <CLIENT-10464>        on Windows-10-10.0.18362, fs encoding utf-8
2020-12-18 12:11:10.823504 +0100  <CLIENT-10464>  Fatal Error: No arguments given
See the rdiff-backup manual page for more information.
```

Linux : 

````
root@1.2.3.4:~$ rdiff-backup -v9
2020-12-18 11:13:17.263691 +0000  <CLIENT-49224>  Fatal Error: No arguments given
See the rdiff-backup manual page for more information.
2020-12-18 11:13:17.250738 +0000  <CLIENT-49224>  Using rdiff-backup version 2.0.5
2020-12-18 11:13:17.250835 +0000  <CLIENT-49224>        with cpython /usr/bin/python3 version 3.5.2
2020-12-18 11:13:17.263591 +0000  <CLIENT-49224>        on Linux-4.4.0-130-generic-x86_64-with-Ubuntu-16.04-xenial, fs encoding utf-8
````

## rdiff-backup call

How did you call rdiff-backup, with which parameters:

```
PS D:\repos\backup_manager> D:\repos\backup_manager\rdiff-backup-2.0.5\rdiff-backup.exe -v9 --remote-schema "C:\Windows\System32\OpenSSH\ssh.exe %s /usr/local/bin/rdiff-backup --server" --include "/var/www**"  --exclude "**" --print-statistics "root@1.2.3.4::/" "D:\Backups\server"
```

## What happened and what did you expect?

I expected the backup run to start.

```
The system cannot find the path specified.
2020-12-18 12:05:13.701586 +0100  <CLIENT-26096>  Fatal Error: Truncated header string (problem probably originated remotely)

Couldn't start up the remote connection by executing

    C:\Windows\System32\OpenSSH\ssh.exe root@1.2.3.4 /usr/local/bin/rdiff-backup --server

Remember that, under the default settings, rdiff-backup must be
installed in the PATH on the remote system.  See the man page for more
information on this.  This message may also be displayed if the remote
version of rdiff-backup is quite different from the local version (2.0.5).
2020-12-18 12:05:13.664580 +0100  <CLIENT-26096>  Using rdiff-backup version 2.0.5
2020-12-18 12:05:13.664580 +0100  <CLIENT-26096>        with cpython D:\repos\backup_manager\rdiff-backup-2.0.5\rdiff-backup.exe version 3.7.7
2020-12-18 12:05:13.665580 +0100  <CLIENT-26096>        on Windows-10-10.0.18362, fs encoding utf-8
2020-12-18 12:05:13.666580 +0100  <CLIENT-26096>  Executing C:\Windows\System32\OpenSSH\ssh.exe root@1.2.3.4 /usr/local/bin/rdiff-backup --server
2020-12-18 12:05:13.682613 +0100  <CLIENT-26096>  Client sending (0): ConnectionRequest: Globals.get with 1 arguments
2020-12-18 12:05:13.682613 +0100  <CLIENT-26096>  Client sending (0): version
```

## More information

The remote server has rdiff-backup 2.0.5 installed using python pip3.

```
PS D:\repos\backup_manager> C:\Windows\System32\OpenSSH\ssh.exe -C root@1.2.3.4
Welcome to Ubuntu 16.04.6 LTS (GNU/Linux 4.4.0-130-generic x86_64)
(.....)
root@3dm-dev-website-vm:~# /usr/local/bin/rdiff-backup --version
rdiff-backup 2.0.5
```

**Comment by ericzolf** [comment]
Just replacing `C:\Windows\System32\OpenSSH\ssh.exe` with `ssh.exe` works. Any other combination, I could think of, 2, 3 or 4 backslashes, replace the backslashes with slashes, remove `C:`, nothing worked, only the pure `ssh.exe`.

**Comment by ericzolf** [comment]
It's strange: a locally compiled version doesn't seem to have the issue, only the "official" version compiled through our pipeline.

**Comment by ericzolf** [comment]
[rdiff-backup-2.0.5-cp37-cp37m-win_amd64.zip](https://github.com/rdiff-backup/rdiff-backup/files/5720005/rdiff-backup-2.0.5-cp37-cp37m-win_amd64.zip)
If you could try the attached file and confirm that it fixes the issue, I would greatly appreciate.

```
$ md5sum rdiff-backup-2.0.5-cp37-cp37m-win_amd64.exe
d04f96629a01762ba46d7a65e3d2db49  rdiff-backup-2.0.5-cp37-cp37m-win_amd64.exe
```

**Comment by xastor** [comment]
I can confirm that when using the attached version of rdiff-backup, without any other changes, I don't have this issue! Everything runs as expected.  Thank you for you speedy intervention!

But the backup does not complete.  The backup stops after a few seconds, with this in the backup.log file : 

```
Starting mirror / to D:/Backups/product-dev
Processing changed file .
Processing changed file var
Processing changed file var/www
Processing changed file var/www/.odbc.ini
Processing changed file var/www/html
Processing changed file var/www/html/.htaccess
Processing changed file var/www/html/19.5.0_2019.04.30_12.40.28_pieter
Processing changed file var/www/html/19.5.0_2019.04.30_12.40.28_pieter/api
Exception 'symbolic link privilege not held' raised of class '<class 'OSError'>':
  File "rdiff_backup\robust.py", line 35, in check_common_error
  File "rdiff_backup\rpath.py", line 156, in copy
  File "rdiff_backup\rpath.py", line 1232, in symlink
```

Could the reason be that rdiff-backup is trying to backup a symbolic link onto a ntfs filesystem?

**Comment by ericzolf** [comment]
Sounds like https://www.scivision.dev/windows-symbolic-link-permission-enable/ to me. We plan to move to Python 3.8 (or 3.9) under Windows so that the issue might go away.

**Comment by xastor** [comment]
Yes, I think this is indeed the underlying cause.  I worked around this by using the `--exclude-symbolic-links` flag for now,

**Comment by ericzolf** [comment]
Back to the initial topic: you wrote in https://github.com/rdiff-backup/rdiff-backup/issues/496#issuecomment-749487815 that using only `ssh.exe` in the remote schema does NOT fix the issue for you. Do you really get the same error message?

**Comment by xastor** [comment]
No, sorry if that was what I said.  It also does not work, but the output is mostly the same, apart from one line.  Instead of this : 

```
The system cannot find the path specified.
```

I get this : 

```
'ssh.exe' is not recognized as an internal or external command,
operable program or batch file.
```

In another surprise discovery, I found that copying the entire `C:\Windows\System32\OpenSSH` folder next to my backup tools and referencing those, works! This is hard to explain...

**Comment by ericzolf** [comment]
Sounds like the `PATH` environment variable has different values depending on the context? I had also the issue that the running SSH service wasn't taking the new value until I had done a reboot of the machine.

**Comment by ericzolf** [comment]
Found it! Check e.g. https://www.samlogic.net/articles/sysnative-folder-64-bit-windows.htm (applies to rdiff-backup compiled 32 bits)

The solution is to use: `--remote-schema "C:\Windows\SysNative\OpenSSH\ssh.exe %s rdiff-backup --server"` (respectively `%%s` if used within a batch file). Notice the `SysNative` instead of `System32`, which rdiff-backup doesn't see when it's compiled in 32 bits.

Let me know if this solves your issue but don't close the issue, I'll close it once I've documented the workaround.

Completely unrelated but while I'm at it, I'll also document the possibility to use something like `--exclude-globbing-file        "%~dpn0.txt"` (where d=Disk, p=(dir)path, n=name(without extension), x=extension) in a batch file as well.
